src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/Integrations/ConsumerConsumeSyncIntegration.cs

```diff
@@ -0,0 +1,96 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using OpenTelemetry.AutoInstrumentation.CallTarget;
+using OpenTelemetry.AutoInstrumentation.DuckTyping;
+using OpenTelemetry.AutoInstrumentation.Instrumentations.Kafka.DuckTypes;
+using OpenTelemetry.Context.Propagation;
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.Kafka.Integrations;
+
+/// <summary>
+/// Kafka consumer sync consume instrumentation
+/// </summary>
+[InstrumentMethod(
+    assemblyName: IntegrationConstants.ConfluentKafkaAssemblyName,
+    typeName: IntegrationConstants.ConsumerTypeName,
+    methodName: IntegrationConstants.ConsumeSyncMethodName,
+    returnTypeName: IntegrationConstants.ConsumeResultTypeName,
+    parameterTypeNames: new[] { ClrNames.Int32 },
+    minimumVersion: IntegrationConstants.MinVersion,
+    maximumVersion: IntegrationConstants.MaxVersion,
+    integrationName: IntegrationConstants.IntegrationName,
+    type: InstrumentationType.Trace)]
+public static class ConsumerConsumeSyncIntegration
+{
+    internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, int timeout)
+    {
+        // Attempting to consume another message marks the end
+        // of a previous message processing
+        KafkaCommon.StopCurrentConsumerActivity();
+
+        return CallTargetState.GetDefault();
+    }
+
+    internal static CallTargetReturn<TResponse> OnMethodEnd<TTarget, TResponse>(TTarget instance, TResponse response, Exception? exception, in CallTargetState state)
+    {
+        IConsumeResult? consumeResult;
+        if (exception is not null && exception.TryDuckCast<IResultException>(out var resultException))
+        {
+            consumeResult = resultException.ConsumerRecord;
+        }
+        else
+        {
+            consumeResult = response == null ? null : response.DuckAs<IConsumeResult>();
+        }
+
+        if (consumeResult is null)
+        {
+            return new CallTargetReturn<TResponse>(response);
+        }
+
+        var propagatedContext = Propagators.DefaultTextMapPropagator.Extract(default, consumeResult, KafkaCommon.MessageHeaderValueGetter);
+        string? spanName = null;
+        if (!string.IsNullOrEmpty(consumeResult.Topic))
+        {
+            spanName = $"{consumeResult.Topic} {MessagingAttributes.Values.ProcessOperationName}";
+        }
+
+        spanName ??= MessagingAttributes.Values.ProcessOperationName;
```

Comment:
The "process" operation was removed in favor of a "deliver" operation for push-based scenarios, and a "receive" operation for pull-based scenarios.

As this instruments a polling method, the operation name should be "receive".

Reply:
@pyohannes 
Apart from adjusting operation name, I think where activity is stopped should be adjusted as well.
`receive` operation is defined to be connected with a consumer requesting one or more messages.
Taking that into account, I think activity created here should be stopped here as well.

Reply:
That's correct, good catch.

Your current logic will result in correct "process" span durations only if the following holds true:

> // Attempting to consume another message marks the end
> // of a previous message processing

However, there are many consumption scenarios where this is not the case. For example, think about fetching messages and then distributing them to different threads for processing.

You should create "Receive" spans here measuring the duration of the `Consume` operation and linking to the context attached to the message.

Reply:
changed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3055/commits/fb2e35bf6a6b09871bec0195dfb2943655521699

Reply:
There is additional section in [OTEP](https://github.com/open-telemetry/oteps/blame/main/text/trace/0220-messaging-semantic-conventions-span-structure.md#L200-L203) that I think applies here:

> For single-message scenarios, and if the "Deliver" or "Receive" spans would be
> root spans of a new trace, the creation context may also be used as a parent on
> those operations in addition to being added as a link. Keeping single-messages
> operations in the same trace can greatly improve the user experience.

Situation described above will often be the case here. 

I noticed this was discussed in a PR updating semantic conventions for messaging, but finally [removed](https://github.com/open-telemetry/semantic-conventions/pull/284#discussion_r1347596683).

Can you confirm it is preferred NOT to use creation context as a parent in this scenario, in addition to linking to it?